spring-integration-core/src/main/java/org/springframework/integration/StaticMessageHeaderAccessor.java

```diff
@@ -113,4 +114,28 @@ public static AcknowledgmentCallback getAcknowledgmentCallback(Message<?> messag
 		return (T) message.getHeaders().get(IntegrationMessageHeaderAccessor.SOURCE_DATA);
 	}
 
+	/**
+	 * Return the method called from the messaging gateway interface or {@code null}
+	 * from the headers in the provided message.
+	 * @param message the message to extract gateway method header.
+	 * @return the method called from the messaging gateway interface.
+	 * @since 5.2
+	 */
+	@Nullable
+	public static Method getGatewayMethod(Message<?> message) {
+		return message.getHeaders().get(IntegrationMessageHeaderAccessor.GATEWAY_METHOD, Method.class);
+	}
```

Comment:
I am wondering how important it is to store reference to the actual method? Reading the docs it appears that the intention is to _know the method name_. Plus having reference to a method exposes the possibility that someone may actually invoke it and consequences and so on. . .

Reply:
The problem there is you can have overloaded methods with the same name and different return types. This was a problem we fixed long ago with the #gatewayMethod SpEL variable.I had similar concerns back then but we decided the benefits outweigh the risks.